firewood/src/merkle/changes.rs

```diff
@@ -0,0 +1,389 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use std::iter::once;
+
+use firewood_storage::{
+    Child, FileIoError, HashedNodeReader, Node, NodeReader, Path, SharedNode, TrieHash,
+    firewood_counter,
+};
+
+use crate::{
+    iter::key_from_nibble_iter,
+    merkle::{Key, PrefixOverlap},
+};
+
+/// Contains all of a node's state that is needed for node comparison in `DiffMerkleNodeStream`.
+/// It includes the nodes full path and its hash if available.
+#[cfg_attr(not(test), expect(dead_code))]
+#[derive(Clone)]
+struct NodeState {
+    path: Path,
+    node: SharedNode,
+    hash: Option<TrieHash>,
+}
+
+impl NodeState {
+    // Creates a NodeState from a `Child`, its pre-path, and trie for reading the node from storage.
+    fn new<T: NodeReader>(pre_path: Path, child: &Child, trie: &T) -> Result<Self, FileIoError> {
+        let node = child.as_shared_node(trie)?;
+        // Create the full path for the node. This is used to compare with the full path
+        // of the current node from the other trie.
+        let mut path = pre_path;
```

Comment:
nit: path -> fullpath

Then you can shorten the comment by removing the first sentence.